src/fields/fp.rs

```diff
@@ -27,6 +33,8 @@ use crate::arithmetic::SqrtTables;
 // Montgomery form; i.e., Fp(a) = aR mod p, with R = 2^256.
 #[derive(Clone, Copy, Eq)]
 #[repr(transparent)]
+#[cfg_attr(feature = "borsh", derive(BorshSerialize, BorshDeserialize))]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
```

Comment:
This is directly serializing the internal Montgomery representation of the field element, which is not part of the public API. Note that `Fp::from_raw` takes a raw integer representation, but _not_ in Montgomery form. It is also unsafe because it allows non-canonical representations of field elements to be deserialized. We should instead have manual implementations that go through `ff::PrimeField::{from_repr, to_repr}` and map the `None` case of `CtOption` to a deserialization error.